crates/ironrdp-dvc-pipe-proxy/src/platform/windows/worker.rs

```diff
@@ -0,0 +1,173 @@
+use std::sync::mpsc;
+
+use ironrdp_core::{ensure_size, Encode, EncodeResult};
+use ironrdp_dvc::{encode_dvc_messages, DvcEncode};
+use ironrdp_pdu::PduResult;
+use ironrdp_svc::{ChannelFlags, SvcMessage};
+
+use crate::platform::windows::error::DvcPipeProxyError;
+use crate::windows::{wait_any, wait_any_with_timeout, Event, MessagePipeServer, Semaphore, WindowsError};
+
+const PIPE_CONNECT_TIMEOUT: u32 = 10_000; // 10 seconds
+const PIPE_WRITE_TIMEOUT: u32 = 3_000; // 3 seconds
+const MESSAGE_BUFFER_SIZE: usize = 64 * 1024; // 64 KiB
```

Comment:
suggestion: For clarity, I recommend to add a suffix such as `_SECS`, OR to use the type `Duration` to avoid unit bugs.